sdk/python/feast/infra/registry_stores/sql.py

```diff
@@ -0,0 +1,413 @@
+            raise FeatureServiceNotFoundException(name, project=project)
+
+    def get_saved_dataset(
+        self, name: str, project: str, allow_cache: bool = False
+    ) -> SavedDataset:
+        pass
+
+    def get_validation_reference(
+        self, name: str, project: str, allow_cache: bool = False
+    ) -> ValidationReference:
+        pass
+
+    def list_entities(self, project: str, allow_cache: bool = False) -> List[Entity]:
+        return self._list_objects(entities, EntityProto, Entity, "entity_proto")
+
+    def delete_entity(self, name: str, project: str, commit: bool = True):
+        with self.engine.connect() as conn:
+            stmt = delete(entities).where(entities.c.entity_id == name)
+            rows = conn.execute(stmt)
+            if rows.rowcount < 1:
+                raise EntityNotFoundException(name, project)
+
+    def delete_feature_view(self, name: str, project: str, commit: bool = True):
+        deleted_count = 0
+        for table in {feature_views, request_feature_views, on_demand_feature_views}:
+            with self.engine.connect() as conn:
+                stmt = delete(table).where(table.c.feature_view_name == name)
+                rows = conn.execute(stmt)
+                deleted_count += rows.rowcount
+        if deleted_count == 0:
+            raise FeatureViewNotFoundException(name, project)
+
+    def delete_feature_service(self, name: str, project: str, commit: bool = True):
+        with self.engine.connect() as conn:
+            stmt = delete(feature_services).where(
+                feature_services.c.feature_service_name == name
+            )
+            rows = conn.execute(stmt)
+            if rows.rowcount < 1:
+                raise FeatureServiceNotFoundException(name, project)
+
+    def get_data_source(
+        self, name: str, project: str, allow_cache: bool = False
+    ) -> DataSource:
+        with self.engine.connect() as conn:
+            stmt = select(data_sources).where(data_sources.c.entity_id == name)
+            row = conn.execute(stmt).first()
+            if row:
+                ds_proto = DataSourceProto.FromString(row["data_source_proto"])
+                return DataSource.from_proto(ds_proto)
+            raise DataSourceObjectNotFoundException(name, project=project)
+
+    def list_data_sources(
+        self, project: str, allow_cache: bool = False
+    ) -> List[DataSource]:
+        return self._list_objects(
+            data_sources, DataSourceProto, DataSource, "data_source_proto"
+        )
+
+    def apply_data_source(
+        self, data_source: DataSource, project: str, commit: bool = True
+    ):
+        return self._apply_object(
+            data_sources, "data_source_name", data_source, "data_source_proto"
+        )
+
+    def apply_feature_view(
+        self, feature_view: BaseFeatureView, project: str, commit: bool = True
+    ):
+        if isinstance(feature_view, FeatureView):
```

Comment:
can u add a comment here that sfvs will be handled as fvs for the time being

Reply:
Sure